rfcs/0093_Issue_Triage.md

```diff
@@ -0,0 +1,309 @@
+
+![New Issue Template](assets/0089/New%20Issue%20Templates.png)
+
+When a new issue is created with the `needs-triage` label, the `triage-bot` will parse the issue’s title or text body (see below for a discussion of options) for label tags. If one or more is found, `triage-bot` will remove `needs-triage` and add the found labels to the issue.
+
+This RFC proposes to modify the :bug: **Bug Report** and :wrench: **Feature Tracking** templates to add instructions encouraging users to self-select which component an issue should belong to. These templates will link to a Markdown doc (checked into TVM) that lists all the label tags and includes a small description for the scope of each one. This will help the users find the appropriate label tags to use.
+
+### Example User Workflow
+
+In these templates, the `triage-bot` will parse the issue’s text body for tags in bullet form.
+
+For example, when creating an issue concerning a bug in `autotvm` , the process would be -
+
+1. The user would select the **Bug Report** template and will see a section shown in body to list the tags they would like to use. It will include a link to the document mentioned above which lists all the label tags.
+2. The user would then add the relevant tags. In this case, they would add `autotvm`. In the case that the issue creator is unsure of which label tags to use, they can simply leave the label tag section blank. This template will add the `needs-triage` label by default which can alert other TVM contributors who can assist in properly sorting the issue.
+3. After filling out the quick summary of the issue, they would post it. It will look like shown in this [example here](https://drive.google.com/file/d/12IHgPFcijcSsbHrY5IY9ACqDlpl2AbP9/view?usp=sharing) and will trigger the `triage-bot` to parse the issue body and add the appropriate label.
+4. Then, the `triage-bot` removes the `needs-triage` label and adds the appropriate label. In this example, it would add `autotvm`.
+
+![TVM Bot](assets/0089/TVM%20bot.png)
+
+If `triage-bot` is unable to find any labels, it will leave the issue alone. The `needs-triage` label will continue to indicate issues which need manual triage. We hope by establishing this category that it encourages more community members to work together to triage issues by consolidating those that need it into a single location.
+
+### Rationale
+
+This will improve quality of the issue tracking and make it easier to drill down on specific pain points for different components. Users will be able to quickly sort the issues by their areas of interests and needs.
+
+Issues can have multiple labels tagged, although we ask community members to exercise their judgement and file issues in as few components as the situation necessitates.
+
+Contributors can propose a new label by posting on the Discuss forum. Requests should include adequate scoping to justify the label, as was done in this doc. Labels can similarly be retired via a proposal on Discuss.
```

Comment:
What if the contributor wants to extend the scope of a label (e.g. when they have recently added another relevant file), would they have to go through forum as well?

Reply:
In general, I think this should be at the discretion of the project--right now, I think that means at the discretion of triagers/committers. For minor extensions of scope (i.e. adding a new file as you say), I think a committer could exercise this discretion independently. For major extensions of scope (i.e. expanding "ir" sub-label to include a new class of IR), this could be posed on the forum.

Additionally, right now there isn't any sort of division of responsibility over the created set of labels. This is covered under Future Work (and in my mind, would be actuated by modfiying CONTRIBUORS.md to state areas of committer/reviewer focus in terms of these labels rather than the somewhat arbitrary set that exist there now). Given all of these factors at play now and that the RFC vote passed, I'd like to suggest we proceed with committer discretion for now. I'd be happy to keep discussing this on a follow-on Discuss thread or RFC PR if that seems problematic to you.

Reply:
Thanks @areusch for getting back on this! I've got no problem with the label scopes being in the discretion of committers, makes sense to me.

I believe my questions about what's the point of a bot triaging tracking issues have not been addressed, but I reckon I was very late with commenting here and that's really a minor issue, so no objections to this having been merged :)